### PR TITLE
Stv Bid Adapter : initial adapter release

### PR DESCRIPTION
**Pull request by theo-stv**
## Type of change
- [x] New bidder adapter  

## Description of change
- contact email: prebid@smartstream.tv
- test parameters for validating bids:

```
var adUnits = [
    {
        code: 'myDiv',
        mediaTypes: {
            banner: {
                sizes: [
                    [300, 300],
                    [300, 600],
                ]
            }
        },
        bids: [
            {
                bidder: "stv",
                params: {
                    placement: 'tst--11141', 
                }
            }
        ]
    },
    {
        code: 'video-slot1',
        mediaTypes: {
            video: {
                playerSize: [640, 480],
                context: 'instream'            }
        },
        bids: [{
            bidder: 'stv',
            params: {
                placement: 'tst--613',
            }
        }]
    }
];
```

### CODE REVIEW

File: modules/stvBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        };
+      }
+
+      payload.pfilter = { ...params };
+      delete payload.pfilter.placement;
+      if (params.bcat !== undefined) { delete payload.pfilter.bcat; }
+      if (params.dvt !== undefined) { delete payload.pfilter.dvt; }
+      if (params.devMode !== undefined) { delete payload.pfilter.devMode; }
+
+      if (mediaTypesInfo[VIDEO] !== undefined) {
```
Comment (ncolletti) [inline modules/stvBidAdapter.js:79]: should this conditional use the `type` result above?

File: modules/stvBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    return validBidRequests.map(bidRequest => {
+      const params = bidRequest.params;
+
+      const placementId = params.placement;
+      const rnd = Math.floor(Math.random() * 99999999999);
+      const referrer = bidderRequest.refererInfo.page;
+      const bidId = bidRequest.bidId;
+      const isDev = params.devMode || false;
+      const pbcode = bidRequest.adUnitCode || false; // div id
+      // const auctionId = bidRequest.auctionId || false;
```
Comment (ncolletti) [inline modules/stvBidAdapter.js:36]: this can be removed if it is not necessary to your adapter

File: modules/stvBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+      if ('gdprApplies' in gdprConsent && typeof gdprConsent.gdprApplies === 'boolean') {
+        gdprParams = `gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
+      } else {
+        gdprParams = `gdpr_consent=${gdprConsent.consentString}`;
+      }
+    }
+
+    if (serverResponses.length > 0 && serverResponses[0].body.userSync) {
+      if (syncOptions.iframeEnabled) {
+        serverResponses[0].body.userSync.iframeUrl.forEach((url) => syncs.push({
```
Comment (ncolletti) [inline modules/stvBidAdapter.js:167]: please add checks for existence of each property in the serverResponse array entry before running each iframe and pixel enabled sync loop

File: modules/stvBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+          alternative: 'prebid_js',
+          _ps: placementId,
+          srw: sizes ? sizes[0].width : 0,
+          srh: sizes ? sizes[0].height : 0,
+          idt: 100,
+          rnd: rnd,
+          ref: referrer,
+          bid_id: bidId,
+          pbver: '$prebid.version$'
+        };
```
Comment (ncolletti) [inline modules/stvBidAdapter.js:70]: seems like all properties other than "_f" are shared in these two conditional blocks. define the object once and adjust the property value within each block

File: modules/stvBidAdapter.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                    context: 'instream'
+                }
+            },
+            bids: [{
+                bidder: 'stv',
+                params: {
+                    placement: '106',
+                    /* // [optional params]
+                        bcat:  "IAB2,IAB4",  // [optional] list of blocked advertiser categories (IAB), comma separated 
+                        floorprice: 1000000, // input min_cpm_micros, CPM in EUR * 1000000
```
Comment (ncolletti) [inline modules/stvBidAdapter.md:52]: Your adapter must support `bid.getFloor` using a `getBidFloor` method to check any globally defined floors first

**Comment by theo-stv** [comment]
@ncolletti I have made the required changes. Please check.

**Review by ncolletti — APPROVED** [review]
changes look good to me!

**Pull request merged**